Deprecate overriding DiscoveryPlugin internals

rjernst opened the pull request:
DiscoveryPlugin allows extending getJoinValidator and
getElectionStrategies. These are implementation details of the system.
This commit deprecates these methods so that plugin authors
are discouraged from overriding them.

rjernst committed:
Deprecation overriding DiscoveryPlugin internals

DiscoveryPlugin allows extending getJoinValidator and
getElectionStrategies. These are implementation details of the system.
This commit deprecates these methods so that plugin authors
are discouraged from overriding them.

rjernst committed:
Deprecation overriding DiscoveryPlugin internals

DiscoveryPlugin allows extending getJoinValidator and
getElectionStrategies. These are implementation details of the system.
This commit deprecates these methods so that plugin authors
are discouraged from overriding them.

elasticsearchmachine commented:
Hi @rjernst, I've created a changelog YAML for you. *Note* that since this PR is labelled `>deprecation`, you need to update the changelog YAML to fill out the extended information sections.

elasticsearchmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

rjernst committed:
Update docs/changelog/88925.yaml

pugnascotia committed:
Add generateStubReleaseNotes task (#88933)

When we feature freeze Elasticsearch, we need to create stub
documentation for the next version. This turns out to be as simple as
running the usual `generateReleaseNotes` task without any inputs.

original-brownbear committed:
Speed up operations on BlobStoreIndexShardSnapshots (#88912)

This fixes a couple of slow points in `BlobStoreIndexShardSnapshots`,
which become performance critical when working with large repositories.

1. Fix `physicalFiles` containing the same `FileInfo` instances repeatedly for every
snapshot that holds the file. Without this fix the map can hold lists as long as the
number of snapshots for the shard for files common to all snapshots of the shard.
Also, only lazy build the map since it's only used during snapshotting and internalize
the logic into `BlobStoreIndexShardSnapshots` so we don't have to bother with wrapping
as unmodifiable.
2. Add efficient copy constructors for all 3 operations on the shard to avoid
expensive looping over all snapshots and their files in many cases.
3. Use list instead of redundant map in deserialization, we weren't using the map
for any deduplication anyways and are safe here thanks to Jackson's duplicate name
detection

benwtrent committed:
[ML] fix BERT and MPNet tokenization bug when handling unicode accents (#88907)

When handling unicode accents, it may have been that BERT tokenizations removed the incorrect characters. This would result in an exceptionally strange result and possibly an error.

closes #88900

grcevski commented on docs/reference/migration/migrate_8_5.asciidoc:
```diff
@@ -20,3 +20,38 @@ coming::[8.5.0]
+The following functionality has been deprecated in {es} 8.5
+and will be removed in a future version.
+While this won't have an immediate impact on your applications,
+we strongly encourage you take the described steps to update your code
```
missing 'to' between you and take.

grcevski reviewed (approved):
LGTM! One minor typo and it seems the YML area needs to be fixed. We possibly want to add an area for the plugins as we'll do bunch of this?

benwtrent committed:
[ML] add sentence-piece unigram tokenizer (#88858)

Add internal unigram tokenizer.

This tokenizer is the same that XLM-Roberta utilizes, along with many other cross-lingual models and tasks.

This does not fully integrate (adding configuration, integrating into nlp tasks, etc.). But instead is just the internal tokenization and some tests showing how it runs with a precompiled charsmap.

rjernst commented:
> We possibly want to add an area for the plugins as we'll do bunch of this?

Yes, @tvernum is going to followup with the docs team to add that. In the mean time, I will use `Java API` here.

mark-vieira committed:
Improve capturing of reaper failure logs

original-brownbear committed:
Make Settings Diffable (#88815)

Making settings diffable so that index metadata diffs
are smaller whenever the metadata changes without a setting change
as well as to make index setting updates over a wider number of
indices faster.

This saves about 3% of CPU time on master and about half that on data nodes
that is just burnt for writing setting strings
when bootstrapping many shards benchmarks benchmarks to 50k indices.

ChrisHegarty committed:
Ensure that the extended socket options TCP_KEEPXXX are available (#88935)

original-brownbear committed:
Bring back lost optimization to building metadata from a diff (#88950)

This was lost again due to another merge conflict.

rjernst committed:
Deprecate network plugins (#88924)

Network plugins provide network implementations. In the past this has
been used for alternatives to netty based networking, using the JDK's
nio. However, nio has now been removed, and it is inadvisable for a
plugin to implement this low level part of the system.
Therefore, this commit marks the NetworkPlugin interface as deprecated.

rjernst committed:
tweak changelog

rjernst committed:
Merge branch 'main' into plugins/deprecate_disco_plugin_internals

rjernst commented:
@elasticsearchmachine run elasticsearch-ci/packaging-tests-unix-sample

rjernst commented:
@elasticsearchmachine run elasticsearch-ci/packaging-tests-unix-sample
